src/firebolt/async_db/connection.py

```diff
@@ -326,25 +322,34 @@ def __init__(
             transport=transport,
             headers={"User-Agent": get_user_agent_header(user_drivers, user_clients)},
         )
-        self.api_endpoint = api_endpoint
-        self.engine_url = engine_url
-        self.database = database
-        self._cursors: List[BaseCursor] = []
-        self._is_closed = False
-
-    def _cursor(self, **kwargs: Any) -> BaseCursor:
-        """
-        Create new cursor object.
-        """
 
+    def cursor(self, **kwargs: Any) -> Cursor:
         if self.closed:
             raise ConnectionClosedError("Unable to create cursor: connection closed.")
 
-        c = self.cursor_class(self._client, self, **kwargs)
+        c = self.cursor_class(client=self._client, connection=self, **kwargs)
         self._cursors.append(c)
         return c
 
-    async def _aclose(self) -> None:
+    # Context manager support
+    async def __aenter__(self) -> Connection:
+        if self.closed:
+            raise ConnectionClosedError("Connection is already closed.")
+        return self
+
+    def _remove_cursor(self, cursor: Cursor) -> None:
```

Comment:
This method, `closed` and `commit` seem to be identical for both classes. Can we move them to the base class

Reply:
I think it makes sense
The less code duplication we have the better IMO